Currently looking into

```
jvm 1    | [08:55:59] ERROR [org.apache.directory.server.wrapper.ApacheDsTanukiWrapper] - Failed to start the service.
jvm 1    | org.apache.directory.api.ldap.model.exception.LdapOtherException: ERR_250_ENTRY_ALREADY_EXISTS ou=system already exists!
jvm 1    |      at org.apache.directory.server.core.api.partition.AbstractPartition.initialize(AbstractPartition.java:94)
jvm 1    |      at org.apache.directory.server.core.DefaultDirectoryService.initialize(DefaultDirectoryService.java:1822)
jvm 1    |      at org.apache.directory.server.core.DefaultDirectoryService.startup(DefaultDirectoryService.java:1251)
jvm 1    |      at org.apache.directory.server.ApacheDsService.initDirectoryService(ApacheDsService.java:364)
jvm 1    |      at org.apache.directory.server.ApacheDsService.start(ApacheDsService.java:197)
jvm 1    |      at org.apache.directory.server.ApacheDsService.start(ApacheDsService.java:153)
jvm 1    |      at org.apache.directory.server.wrapper.ApacheDsTanukiWrapper.start(ApacheDsTanukiWrapper.java:175)
jvm 1    |      at org.tanukisoftware.wrapper.WrapperManager$12.run(WrapperManager.java:2788)
jvm 1    | Caused by: org.apache.directory.api.ldap.model.exception.LdapOtherException: ERR_250_ENTRY_ALREADY_EXISTS ou=system already exists!
jvm 1    |      at org.apache.directory.server.core.api.partition.AbstractPartition.initialize(AbstractPartition.java:94)
jvm 1    |      at org.apache.directory.server.core.shared.partition.DefaultPartitionNexus.addContextPartition(DefaultPartitionNexus.java:809)
jvm 1    |      at org.apache.directory.server.core.shared.partition.DefaultPartitionNexus.doInit(DefaultPartitionNexus.java:227)
jvm 1    |      at org.apache.directory.server.core.api.partition.AbstractPartition.initialize(AbstractPartition.java:89)
jvm 1    |      ... 7 more
jvm 1    | Caused by: org.apache.directory.api.ldap.model.exception.LdapEntryAlreadyExistsException: ERR_250_ENTRY_ALREADY_EXISTS ou=system already exists!
jvm 1    |      at org.apache.directory.server.core.partition.impl.btree.AbstractBTreePartition.add(AbstractBTreePartition.java:698)
jvm 1    |      at org.apache.directory.server.core.partition.impl.btree.jdbm.JdbmPartition.doInit(JdbmPartition.java:522)
jvm 1    |      at org.apache.directory.server.core.api.partition.AbstractPartition.initialize(AbstractPartition.java:89)
jvm 1    |      ... 10 more
```

See also:

 * https://github.com/greggigon/apacheds/issues/4
 * https://issues.apache.org/jira/browse/DIRSERVER-1954
 * etc.

--- a/Dockerfile
+++ b/Dockerfile
@@ -13,7 +13,7 @@ ENV APACHEDS_DATA /var/lib/apacheds
 ENV APACHEDS_USER apacheds
 ENV APACHEDS_GROUP apacheds
 
-RUN ln -s ${APACHEDS_DATA}-${APACHEDS_VERSION} ${APACHDS_DATA}
+RUN ln -s ${APACHEDS_DATA}-${APACHEDS_VERSION} ${APACHEDS_DATA}
 VOLUME ${APACHEDS_DATA}
 
 RUN echo 'debconf debconf/frontend select Noninteractive' | debconf-set-selections \
@@ -39,11 +39,9 @@ EXPOSE 10389 10636 60088 60464 8080 8443
 ENV APACHEDS_INSTANCE default
 ENV APACHEDS_BOOTSTRAP /bootstrap
 
-ENV APACHEDS_SCRIPT run.sh
-ENV APACHEDS_CMD /${APACHEDS_SCRIPT}
-ADD scripts/${APACHEDS_SCRIPT} ${APACHEDS_CMD}
-RUN chown ${APACHEDS_USER}:${APACHEDS_GROUP} ${APACHEDS_CMD} \
-    && chmod u+rx ${APACHEDS_CMD}
+ADD scripts/run.sh /run.sh
+RUN chown ${APACHEDS_USER}:${APACHEDS_GROUP} /run.sh \
+    && chmod u+rx /run.sh
 
 ADD instance/* ${APACHEDS_BOOTSTRAP}/conf/
 ADD ome.ldif ${APACHEDS_BOOTSTRAP}/

--- a/Dockerfile
+++ b/Dockerfile
@@ -61,7 +61,7 @@ ADD bin/ldapmanager /usr/local/bin/ldapmanager
 #############################################
 
 # Correct for hard-coded INSTANCES_DIRECTORY variable
-RUN sed -i 's#/var/lib/apacheds-2.0.0-M24#/var/lib/apacheds#' /opt/apacheds-${APACHEDS_VERSION}/bin/apacheds
+RUN sed -i "s#/var/lib/apacheds-${APACHEDS_VERSION}#/var/lib/apacheds#" /opt/apacheds-${APACHEDS_VERSION}/bin/apacheds
 
 
 RUN curl -L -o /usr/local/bin/dumb-init \

--- a/README.md
+++ b/README.md
@@ -13,7 +13,7 @@ The project sources can be found on [GitHub](https://github.com/openmicroscopy/a
 
 ## Installation
 
-The folder */var/lib/apacheds* contains the runtime data and thus has been defined as a volume. A [volume container](https://docs.docker.com/userguide/dockervolumes/) could be used for that. The image uses exactly the file system structure defined by the [ApacheDS documentation](https://directory.apache.org/apacheds/advanced-ug/2.2.1-debian-instance-layout.html).
+The folder */var/lib/apacheds* contains the runtime data and thus has been defined as a volume. The image uses exactly the file system structure defined by the [ApacheDS documentation](https://directory.apache.org/apacheds/advanced-ug/2.2.1-debian-instance-layout.html).
 
 The container can be started issuing the following command:
 

--- a/scripts/run.sh
+++ b/scripts/run.sh
@@ -9,6 +9,7 @@
 # APACHEDS_GROUP
 
 APACHEDS_INSTANCE_DIRECTORY=${APACHEDS_DATA}/${APACHEDS_INSTANCE}
+PIDFILE="${APACHEDS_INSTANCE_DIRECTORY}/run/apacheds-${APACHEDS_INSTANCE}.pid"
 
 # When a fresh data folder is detected then bootstrap the instance configuration.
 if [ ! -d ${APACHEDS_INSTANCE_DIRECTORY} ]; then
@@ -18,7 +19,6 @@ if [ ! -d ${APACHEDS_INSTANCE_DIRECTORY} ]; then
 fi
 
 cleanup(){
-    PIDFILE="${APACHEDS_INSTANCE_DIRECTORY}/run/apacheds-${APACHEDS_INSTANCE}.pid"
     if [ -e "${PIDFILE}" ];
     then
         echo "Cleaning up ${PIDFILE}"
@@ -26,14 +26,16 @@ cleanup(){
     fi
 }
 
+trap cleanup EXIT
+cleanup
+
+/opt/apacheds-${APACHEDS_VERSION}/bin/apacheds start ${APACHEDS_INSTANCE}
+sleep 2  # Wait on new pid
+
 shutdown(){
     echo "Shutting down..."
     /opt/apacheds-${APACHEDS_VERSION}/bin/apacheds stop ${APACHEDS_INSTANCE}
 }
 
-trap shutdown TERM
-trap cleanup EXIT
-cleanup
-
-# Execute the server in console mode and not as a daemon.
-/opt/apacheds-${APACHEDS_VERSION}/bin/apacheds repair ${APACHEDS_INSTANCE}
+trap shutdown INT TERM
+tail --pid=$(cat $PIDFILE) -f /dev/null

--- a/scripts/run.sh
+++ b/scripts/run.sh
@@ -38,4 +38,4 @@ shutdown(){
 }
 
 trap shutdown INT TERM
-tail --pid=$(cat $PIDFILE) -f /dev/null
+tail -n 0 --pid=$(cat $PIDFILE) -f ${APACHEDS_INSTANCE_DIRECTORY}/log/apacheds.log

